crates/bevy_ecs/src/query/access.rs

```diff
@@ -577,6 +734,33 @@ impl<T: SparseSetIndex> Access<T> {
     pub fn archetypal(&self) -> impl Iterator<Item = T> + '_ {
         self.archetypal.ones().map(T::get_sparse_set_index)
     }
+
+    /// Returns an iterator over the component IDs that this `Access` either
+    /// reads and writes or cannot read or write, depending on the value of
+    /// [`Access::component_reads_and_writes_inverted`].
+    ///
+    /// Because this method depends on internal implementation details of
```

Comment:
I think this method could be made safe to use if it returned an `enum` with variants representing "these components" and "everything except for these components".